pkg/bundle/bundle_test.go

```diff
@@ -124,3 +126,62 @@ func TestMediaTypeString(t *testing.T) {
 		})
 	}
 }
+
+func Test_BundleValidation(t *testing.T) {
+	tests := []struct {
+		name    string
+		bundle  *ProtobufBundle
+		errMsg  string
+		wantErr bool
+	}{
+		{
+			name: "Empty verification material",
+			bundle: &ProtobufBundle{
+				Bundle: &protobundle.Bundle{
+					MediaType:            "application/vnd.dev.sigstore.bundle+json;version=0.3",
+					VerificationMaterial: &protobundle.VerificationMaterial{},
+					Content:              nil,
+				},
+			},
+			errMsg:  "bundle validation failed: missing required field: verification_material.public_key",
+			wantErr: true,
+		},
+		{
+			name: "No verification material",
+			bundle: &ProtobufBundle{
+				Bundle: &protobundle.Bundle{
+					MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.3",
+					Content:   nil,
+				},
+			},
+			errMsg:  "bundle validation failed: missing required field: verification_material",
+			wantErr: true,
+		},
+		{
+			name: "Valid protobuf bundle",
+			bundle: &ProtobufBundle{
+				Bundle: &protobundle.Bundle{
+					MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.3",
+					Content:   nil,
+					VerificationMaterial: &protobundle.VerificationMaterial{
+						Content: &protobundle.VerificationMaterial_PublicKey{
+							PublicKey: &v1.PublicKeyIdentifier{},
+						},
+						TimestampVerificationData: &protobundle.TimestampVerificationData{},
+					},
+				},
+			},
+			errMsg:  "",
+			wantErr: false,
+		},
+	}
```

Comment:
This test case fails because `public_key` is present and `x509_certificate_chain` is not present, when only one has to be present